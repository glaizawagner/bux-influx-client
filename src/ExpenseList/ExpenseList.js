import React, { Component } from 'react';
import ExpenseItem from '../ExpenseItem/ExpenseItem';
import BuxInfluxContext from  '../BuxInfluxContext'

export default class ExpenseList extends Component {
    static defaultProps = {
        expenses: []
      }

      static contextType = BuxInfluxContext;
      
    render() {
        const { expenses } = this.context;
        return(
            <section className='ExpenseList'>
                <h2>Expense</h2>
                <ul className="ExpenseList__list" aria-live='polite'>
                    {expenses.map(exp => 
                        <ExpenseItem 
                            key={exp.id} 
                            {...exp} 
                        />
                    )}
                </ul>
            </section>    
        )
    }
}

