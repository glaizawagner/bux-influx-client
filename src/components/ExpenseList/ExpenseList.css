.expName {
    color: #ff5033;
    text-transform: uppercase;
    display: block;
    text-align: center;
    font-size: 2.5rem;
    font-weight: 900;
    margin-top: 3rem;
    word-spacing: 2em;
    border-top: 2px solid black;
    border-bottom: 2px solid black;
    padding: 0;
    letter-spacing: 1rem;

}

.ExpenseList__list {
    text-align: left;
    margin: 2% 5%;
    list-style: none;
    /* border-top: 2px solid black;
    border-bottom: 2px solid black; */
}

@media only screen and (min-width: 37.5em) and (max-width: 56.25em) {
    .expName {
        margin-top: 3rem;
        margin-left: 10%;
        margin-right: 10%;
    }

    .ExpenseList__list {
        margin: 0 10%;
    }
}

@media only screen and (min-width: 56.25em) and (max-width: 75em) {
    .expName {
        margin-top: 3rem;
        margin-left: 10%;
        margin-right: 10%;
    }

    .ExpenseList__list {
        margin: 0 10%;
    }
}

@media only screen and (min-width: 75em) and (max-width: 105em) {
    .expName {
        margin-top: 3rem;
        margin-left: 20%;
        margin-right: 20%;
    }

    .ExpenseList__list {
        margin: 0 20%;
    }
}