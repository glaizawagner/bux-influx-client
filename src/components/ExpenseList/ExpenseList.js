import React, { Component } from 'react';
import ExpenseItem from '../../components/ExpenseItem/ExpenseItem';
import BuxinfluxContext from  '../../contexts/BuxinfluxContext'
import helpers from '../../components/helpers/helpers';
import './ExpenseList.css';

export default class ExpenseList extends Component {
    static defaultProps = {
        expenses: []
    }

      static contextType = BuxinfluxContext;
      
    render() {
        const { created, expenses } = this.context;
        return(
            <section className='ExpenseList'>
                <span className="expName">Expenses</span>
                <ul className="ExpenseList__list" aria-live='polite'>
                    {expenses.map( (expense,idx) => 
                        (helpers.formatDate(expense.date_created) === helpers.formatDate(created)) ? <ExpenseItem key={idx} eid={idx} {...expense} /> : ''
                    )}
                </ul>
            </section>    
        );
    }
}

