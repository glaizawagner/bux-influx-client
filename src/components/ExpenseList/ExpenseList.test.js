import React from 'react';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer'
import ExpenseList from './ExpenseList';
import BuxinfluxContext from '../../contexts/BuxinfluxContext';

const contextValue = {
    expenses: [
        {
            eid: 1,
            date_created: '2019-11-12T16:28:32.615Z',
            type: 'exp',
            description: 'Salary',
            value: '3500.00',
            percentage: '17.00',
            user_id: 1
        },
        {
            eid: 2,
            date_created: '2019-11-12T16:28:32.615Z',
            type: 'exp',
            description: 'Other Expenses',
            value: '1500.00',
            percentage: '15.00',
            user_id: 2
        }
    ]
}

describe('Expense List  component', () => {
    it('renders UI expected', () => {
        const tree = renderer
        .create(<BuxinfluxContext.Provider value={contextValue}>
                <ExpenseList />
                </BuxinfluxContext.Provider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
    
    });
    
    // enzyme doesn't yet support React.createContext
    it.skip('renders Expense in ul for each in array', () => {
        const props = {
            match: {
                params: {
                    eid: 'THIS_EXPENSE_ID'
                }
            }
        }

        const ul = (<ExpenseList {...props} />, contextValue)
            .find('ul')
            expect(toJson(ul)).toMatchSnapshot()
    })

});