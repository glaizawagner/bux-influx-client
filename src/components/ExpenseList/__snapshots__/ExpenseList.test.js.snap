// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense List  component renders UI expected 1`] = `
<div
  className="IncomeList"
>
  <section
    className="ExpenseList"
  >
    <span
      className="expName"
    >
      Expenses
    </span>
    <ul
      aria-live="polite"
      className="ExpenseList__list"
    >
      <li
        className="ExpenseItem"
      >
        <div
          className="ExpenseItem__row"
        >
          <span>
            <span
              className="items"
            >
              Nov 12, 2019
               
            </span>
            <span
              className="items"
            >
              Salary
               
            </span>
            <span
              className="items"
            >
              3500.00
              <span
                className="items"
              >
                17.00
                 
              </span>
            </span>
            <button
              className="ExpenseItem__btn"
              onClick={[Function]}
            >
              <svg
                className="timesIcon"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<path d=\\"M256 8C119 8 8 119 8 256s111 248 248 248s248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200s-89.5 200-200 200zm101.8-262.2L295.6 256l62.2 62.2c4.7 4.7 4.7 12.3 0 17l-22.6 22.6c-4.7 4.7-12.3 4.7-17 0L256 295.6l-62.2 62.2c-4.7 4.7-12.3 4.7-17 0l-22.6-22.6c-4.7-4.7-4.7-12.3 0-17l62.2-62.2l-62.2-62.2c-4.7-4.7-4.7-12.3 0-17l22.6-22.6c4.7-4.7 12.3-4.7 17 0l62.2 62.2l62.2-62.2c4.7-4.7 12.3-4.7 17 0l22.6 22.6c4.7 4.7 4.7 12.3 0 17z\\" fill=\\"currentColor\\"/>",
                  }
                }
                focusable={false}
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "transform": "rotate(360deg)",
                  }
                }
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              />
            </button>
          </span>
        </div>
      </li>
      <li
        className="ExpenseItem"
      >
        <div
          className="ExpenseItem__row"
        >
          <span>
            <span
              className="items"
            >
              Nov 12, 2019
               
            </span>
            <span
              className="items"
            >
              Other Expenses
               
            </span>
            <span
              className="items"
            >
              1500.00
              <span
                className="items"
              >
                15.00
                 
              </span>
            </span>
            <button
              className="ExpenseItem__btn"
              onClick={[Function]}
            >
              <svg
                className="timesIcon"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<path d=\\"M256 8C119 8 8 119 8 256s111 248 248 248s248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200s-89.5 200-200 200zm101.8-262.2L295.6 256l62.2 62.2c4.7 4.7 4.7 12.3 0 17l-22.6 22.6c-4.7 4.7-12.3 4.7-17 0L256 295.6l-62.2 62.2c-4.7 4.7-12.3 4.7-17 0l-22.6-22.6c-4.7-4.7-4.7-12.3 0-17l62.2-62.2l-62.2-62.2c-4.7-4.7-4.7-12.3 0-17l22.6-22.6c4.7-4.7 12.3-4.7 17 0l62.2 62.2l62.2-62.2c4.7-4.7 12.3-4.7 17 0l22.6 22.6c4.7 4.7 4.7 12.3 0 17z\\" fill=\\"currentColor\\"/>",
                  }
                }
                focusable={false}
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "transform": "rotate(360deg)",
                  }
                }
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              />
            </button>
          </span>
        </div>
      </li>
    </ul>
  </section>
</div>
`;
